app/forms/base_work_form.rb

```diff
@@ -197,10 +197,23 @@ def save!(*)
   def save_model
     super
     # For a zipfile, both the existing files and the zipfile will be attached. These will be removed during unzip.
+
+    # If the previous version has > 250 files and the upload_type is browser then copy the files from previous version.
```

Comment:
i've never had the deepest understanding of how file attachments are modeled/handled in H2, and since i don't really understand the root cause of the issue here or why the fix works, could i ask for a little explanation in this comment about why this is only needed when there are more than `max_upload_files` attachments?

also, assuming my understanding is correct that only the database rows are being copied, could the comment be tweaked to indicate that?  (e.g. "copy the attached_files db rows from previous version")

looking at `BaseWorkForm`, i want to ask something like "would it be more idiomatic/maintainable to do this duplication in `AttachedFilesPopulator`?", but given my (lack of) facility with ReForm, i'm not even sure that is a sensible question.

Reply:
See additional comments which should address most of your questions.

Perhaps this could be done in AttachedFilesPopulator. But I look at all of this as an eyesore until we get rid of reform, so unclear which is best.